lib/client/api.go

```diff
@@ -1646,21 +1646,36 @@ func (tc *TeleportClient) ConnectToNode(ctx context.Context, clt *ClusterClient,
 	mfaCancel()
 	directCancel()
 
+	rootCluster, err := tc.RootClusterName(ctx)
+	if err != nil {
+		rootCluster = tc.SiteName
+	}
+	return nil, trace.Wrap(ChooseConnectionError(directErr, mfaErr, rootCluster, nodeDetails.Cluster))
+}
+
+// ChooseConnectionError determines which error from connecting to a host simultaneously
+// with and without MFA should be returned to the user.
+func ChooseConnectionError(directErr, mfaErr error, rootCluster, targetCluster string) error {
 	switch {
 	// No MFA errors, return any errors from the direct connection
 	case mfaErr == nil:
-		return nil, trace.Wrap(directErr)
+		return trace.Wrap(directErr)
+	// Any access denied errors from a direct connection to a node in another cluster take
+	// precedence over mfa errors. Since the root cluster is always used for MFA we don't
+	// want to show users misleading errors if they don't have any devices in the root cluster.
+	case trace.IsAccessDenied(directErr) && rootCluster != targetCluster:
+		return trace.Wrap(directErr)
```

Comment:
sometimes MFA really is required for remote nodes though, won't this hide the more interesting MFA errors for all remote nodes?